pkg/storegateway/bucket.go

```diff
@@ -1138,26 +1158,23 @@ func (s *BucketStore) streamingSeriesSetForBlocks(
 		})
 	}
 
-	begin := time.Now()
 	err := g.Wait()
 	if err != nil {
 		return nil, nil, err
 	}
 
-	getAllDuration := time.Since(begin)
 	stats.update(func(stats *queryStats) {
 		stats.blocksQueried = len(batches)
-		stats.getAllDuration = getAllDuration
 	})
-	s.metrics.seriesGetAllDuration.Observe(getAllDuration.Seconds())
+	s.metrics.streamingSeriesRequestDurationByStage.WithLabelValues("expand_postings").Observe(time.Since(begin).Seconds())
```

Comment:
I think this slightly changes the semantics of the measurement. Previously we'd measure each individual `ExpandedPostings` call. Now we measure effectively the longest one. I think we should still measure this for each individual `ExpandedPostings` call so we have higher resolution on the histograms. WDYT?

Reply:
`streamingSeriesRequestDurationByStage` is used to track `cortex_bucket_store_series_request_stage_duration_seconds`. The idea of this metric is exactly to have a per-request tracking, so that when you compute the p99 or average, what you get is the split down of a `Series()` request timing.

I don't see much value into measuring individual expanded postings, but the whole time spend doing expanded postings for a given request.

Reply:
i see your point; same reply as in https://github.com/grafana/mimir/pull/3966#discussion_r1072399006